### PR TITLE
test: add coverage tooling, fix flaky tests, add frontend component tests

### LINKED ISSUE #237: Improve test suite: coverage, flaky fixes, dedup, new component tests
## Context

The ARCHITECTURE_REVIEW.md identified significant gaps in test infrastructure. This issue tracks implementing all recommendations from Section 4 (Test Suite Recommendations).

### Backend (Python)
- Add `pytest-cov` to server and client dev dependencies
- Add coverage flags to Makefile test targets and CI jobs
- Add `test-shared` CI job for the shared package
- Fix flaky JWT expiry test (replace `time.sleep` with pre-expired token)
- Fix flaky rate limiter test (replace `time.sleep` with mocked `time.monotonic`)
- Deduplicate validation tests (remove client copies that duplicate shared tests)
- Create shared test helpers (`helpers.py`) and factories (`factories.py`)
- Add error-path API tests (non-zip upload, duplicate publish conflict)
- Extend domain test coverage (tracker partial failure, eval conflict, stale JWT)

### Frontend (TypeScript)
- Add `@vitest/coverage-v8` for coverage measurement
- Create shared test utilities (`test/helpers.tsx`) with factories and `renderWithRouter`
- Add SkillDetailPage tests (13 tests)
- Add AskModal tests (13 tests)
- Add HomePage tests (10 tests)
- Refactor SkillsPage tests to use shared factories

## Deferred
- DB integration tests against real Postgres (separate PR)
- `pytest-xdist` for parallel execution
- Property-based testing with `hypothesis`
- Playwright E2E tests

### PR DESCRIPTION
**Pull request by lfiaschi**
## Summary

- Add **pytest-cov** to server/client with coverage flags in Makefile and CI (`--cov-fail-under=50`)
- Add **test-shared** CI job for the `dhub-core` package
- Fix **2 flaky tests**: JWT expiry (pre-expired token vs `time.sleep`) and rate limiter window (`mock time.monotonic` vs `time.sleep`)
- **Deduplicate** client validation tests that duplicated shared tests
- Create **server test factories** (`factories.py`) and **helpers** (`helpers.py`) for reusable test data
- Add **error-path API tests**: non-zip upload → 422, duplicate publish → 409
- Extend **domain test coverage**: tracker partial failure, eval conflict, stale JWT membership
- Add **@vitest/coverage-v8** for frontend coverage measurement
- Create **frontend test helpers** (`test/helpers.tsx`) with factories and `renderWithRouter`
- Add **36 new frontend component tests**: SkillDetailPage (13), AskModal (13), HomePage (10)
- Refactor SkillsPage tests to use shared `makeSkill` factory

**Coverage results:**
- Server: 69% (800 tests)
- Client: 73% (243 tests)
- Shared: 60 tests pass
- Frontend: 70 tests across 7 files

## Test plan

- [x] `make test-server` — 800 passed, coverage visible
- [x] `make test-client` — 243 passed, coverage visible
- [x] `make test-shared` — 60 passed
- [x] `make test-frontend` — 70 passed (7 test files)
- [x] `make lint` — ruff + tsc + eslint clean
- [x] No `time.sleep` in test files (only mocked references in crawler tests)
- [ ] CI passes all jobs including new `test-shared`

Closes #237

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/tests/factories.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    Returns a plain dict since User is a simple frozen dataclass with
+    only a few fields, and tests often just need the raw values.
+    """
+    defaults: dict[str, object] = {
+        "id": DEFAULT_USER_ID,
+        "github_id": "gh-42",
+        "username": "testuser",
+    }
+    defaults.update(overrides)
+    return defaults
```
[the remark this answers is not in the document]
Comment (lfiaschi) [reply server/tests/factories.py]: Confirmed — removed `make_user()` in cced1c2.
